### PR TITLE
voltage: Update description for voltage service group

### PR DESCRIPTION
**Pull request by yeongjoshua**
Update some description for voltage get attribute
and get supported levels descrption.

### CODE REVIEW

File: src/srvgrp-voltage.adoc [hunk 1]
```diff
@@ -144,16 +144,17 @@ formats. Additional voltage formats can be accommodated in the future if require
 
 	0b1: Voltage domain is always-on, voltage value can be changed in the supported voltage range.
 !===
-| 2	|NUM_LEVELS	 | uint32	| Number of voltage levels supported by 
-domain. Some values are dependent on the VOLTAGE_FORMAT.
+| 2	|NUM_LEVELS	 | uint32	| Number of voltage levels (in word)
```
Comment (pathakraul) [inline src/srvgrp-voltage.adoc:147]: `in word` can be `word-size` ?

File: src/srvgrp-voltage.adoc [hunk 2]
```diff
@@ -144,16 +144,17 @@ formats. Additional voltage formats can be accommodated in the future if require
...
+supported by domain. Values are dependent on the VOLTAGE_FORMAT. Get the
+voltage level format in VOLT_GET_SUPPORTED_LEVELS.
```
Comment (pathakraul) [inline src/srvgrp-voltage.adoc:149]: This line is little incorrect , voltage level format is known from `VOLT_GET_ATTRIBUTES`,  and `VOLT_GET_SUPPORTED_LEVELS` gives you levels not format

File: src/srvgrp-voltage.adoc [hunk 3]
```diff
@@ -144,16 +144,17 @@ formats. Additional voltage formats can be accommodated in the future if require
...
 [cols="2,5a"]
 !===
 ! *Value* 	!  *Description*
 ! 1		! When VOLTAGE_FORMAT=`0x0`
 ! 3		! When VOLTAGE_FORMAT=`0x1`
-! N		! When VOLTAGE_FORMAT=`0x2` or `0x3`. Based on the format here
-each item can be a single voltage value or tuple of values. Check the VOLTAGE_FORMAT
-field in FLAGS.
+! 4N		! When VOLTAGE_FORMAT=`0x2` (Multiple of 4 voltage level word)
```
Comment (pathakraul) [inline src/srvgrp-voltage.adoc:155]: Shall we include one more bit for VOLTAGE_FORMAT for any more formats in future, we already have some in reserved. @lftan?

Comment (lftan) [reply src/srvgrp-voltage.adoc:155]: yes, we can add one more bit for format.

File: src/srvgrp-voltage.adoc [hunk 4]
```diff
@@ -144,16 +144,17 @@ formats. Additional voltage formats can be accommodated in the future if require
 
 	0b1: Voltage domain is always-on, voltage value can be changed in the supported voltage range.
 !===
-| 2	|NUM_LEVELS	 | uint32	| Number of voltage levels supported by 
-domain. Some values are dependent on the VOLTAGE_FORMAT.
+| 2	|NUM_LEVELS	 | uint32	| Number of voltage levels (in word)
+supported by domain. Values are dependent on the VOLTAGE_FORMAT. Get the
```
Comment (lftan) [inline src/srvgrp-voltage.adoc:148]: grammar fix, "supported by the domain"

Comment (lftan) [reply src/srvgrp-voltage.adoc:148]: grammar fix.

"Get  the  voltage  levels with VOLT_GET_SUPPORTED_LEVELS."

File: src/srvgrp-voltage.adoc (hunk 3 above)
Comment (lftan) [reply src/srvgrp-voltage.adoc:155]: plural for words.

"(Multiple of 4 voltage level **words**)"

File: src/srvgrp-voltage.adoc [hunk 5]
```diff
@@ -144,16 +144,17 @@ formats. Additional voltage formats can be accommodated in the future if require
...
 !===
 ! *Value* 	!  *Description*
 ! 1		! When VOLTAGE_FORMAT=`0x0`
 ! 3		! When VOLTAGE_FORMAT=`0x1`
-! N		! When VOLTAGE_FORMAT=`0x2` or `0x3`. Based on the format here
-each item can be a single voltage value or tuple of values. Check the VOLTAGE_FORMAT
-field in FLAGS.
+! 4N		! When VOLTAGE_FORMAT=`0x2` (Multiple of 4 voltage level word)
+! N		! When VOLTAGE_FORMAT=`0x3` (Multiple of 1 depending number of
+discrete voltage level)
```
Comment (lftan) [inline src/srvgrp-voltage.adoc:157]: "voltage levels"

Comment (lftan) [reply src/srvgrp-voltage.adoc:157]: or below is better?

"When VOLTAGE_FORMAT=0x3 (NUM_LEVELS is the number of discrete voltage levels)"

File: src/srvgrp-voltage.adoc [hunk 6]
```diff
@@ -201,42 +202,46 @@ all the voltage levels.
...
+! *Word*	! *Name*	! *Description*
+! 0		! min_uV	! Lower boundary of voltage level (uV)
+! 1		! min_sel	! Lowest selector for range
+! 2		! max_sel	! Highest selector for range
+! 3		! step_uV	! Step size (uV)
+!===
+
+[cols="1,1,5"]
+!===
+3+! *Format: Discrete (VOLT_GET_ATTRIBUTES.FLAGS[31:30]: `3`)*
```
Comment (lftan) [inline src/srvgrp-voltage.adoc:237]: remove ` between 3

File: src/srvgrp-voltage.adoc [hunk 7]
```diff
@@ -201,42 +202,46 @@ all the voltage levels.
...
+! *Word*	! *Name*	! *Description*
+! 0		! level_uV	! Discrete voltage level (uV)
```
Comment (lftan) [inline src/srvgrp-voltage.adoc:239]: choose use "level_uV" or "volt_uV", consistent for discrete and fixed voltage format.

File: src/srvgrp-clock.adoc [hunk 8]
```diff
@@ -146,16 +146,17 @@ the future, as needed.
...
-	CLOCK_RATE[2] = step_Hz   (Step between two successive rates)
+----
+0b00: Discrete Format
+Each element in the CLOCK_RATE array is a supported discrete clock rate value packed in ascending order. Each rate is in Hertz.
+
+0b01: Linear Range
+The CLOCK_RATE array contains the triplet (min_Hz, max_Hz, step_Hz). 
+Each item in the triplet is a clock rate value.
+CLOCK_RATE[0] = min_Hz  (lowest physical rate that the clock can synthesize)
+CLOCK_RATE[1] = max_Hz   (highest physical rate that the clock can synthesize)
```
Comment (lftan) [inline src/srvgrp-clock.adoc:157]: align the "("

File: src/srvgrp-clock.adoc [hunk 9]
```diff
@@ -146,16 +146,17 @@ the future, as needed.
...
+CLOCK_RATE[2] = step_Hz   (Step between two successive rates)
+----
 ! [29:0]	! _Reserved_
 !===
 | 2	|NUM_RATESS	 	| uint32	| Number of clock rates of type depending on CLOCK_FORMAT.
```
Comment (lftan) [inline src/srvgrp-clock.adoc:162]: Typo for "NUM_RATESS", also fix it.

File: src/srvgrp-voltage.adoc [hunk 10]
```diff
@@ -128,32 +128,35 @@ formats. Additional voltage formats can be accommodated in the future if require
...
-
-	0b01: Simple-linear range containing single triplet (min_uV, max_uV, step_uV).
+! [31:29]	! VOLTAGE_FORMAT
+----
+0b000: Fixed voltage
+	
+0b001: Simple-linear range containing single triplet (min_uV, max_uV, step_uV).
 	
-	0b10: Multi-linear range containing multiple linear ranges of type `0x1` where each range contains (min_uV, min_sel, max_sel, step_uV).
+0b010: Multi-linear range containing multiple linear ranges of type `0x1` where each range contains (min_uV, min_sel, max_sel, step_uV).
```
Comment (lftan) [inline src/srvgrp-voltage.adoc:137]: generated PDF still show "`0x1`",  check `

File: src/srvgrp-voltage.adoc [hunk 11]
```diff
@@ -128,32 +128,35 @@ formats. Additional voltage formats can be accommodated in the future if require
...
-
-	0b01: Simple-linear range containing single triplet (min_uV, max_uV, step_uV).
+! [31:29]	! VOLTAGE_FORMAT
+----
+0b000: Fixed voltage
+	
+0b001: Simple-linear range containing single triplet (min_uV, max_uV, step_uV).
 	
-	0b10: Multi-linear range containing multiple linear ranges of type `0x1` where each range contains (min_uV, min_sel, max_sel, step_uV).
+0b010: Multi-linear range containing multiple linear ranges of type '0b010' where each range contains (min_uV, min_sel, max_sel, step_uV).
```
Comment (lftan) [inline src/srvgrp-voltage.adoc:137]: i think no need mentin '0b010' again in the description, simple linear no mention again too.

**Pull request merged**